### PR TITLE
feat(sdk): implement document sum/average aggregation FFI (DOC-13/14)

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
`dash_sdk_document_sum` and `dash_sdk_document_average` were stubs returning `NotImplemented`, citing *"blocked upstream on grovedb PR 670"*. **That block is stale.** The pinned grovedb (`fc81498`) already exposes `verify_aggregate_sum_query` / `verify_aggregate_count_and_sum_query[_per_key]`, and the entire Rust stack below the FFI is complete:
- **rs-drive** — `drive_document_sum_query` / `drive_document_count_and_sum_query` dispatchers + executors, and `verify/document_sum/` proof verifiers.
- **rs-sdk** — `impl Fetch for DocumentSplitSums` / `DocumentSplitAverages`.

This left the two FFI entry points (DOC-13 sum / DOC-14 average) as the only gap.

## What was done?
Implemented both FFI functions as mirrors of the already-working `dash_sdk_document_count`:
- Build the `DocumentQuery` from the same JSON params (`where` / `order_by` / `group_by` / `limit`), selecting `SelectProjection::sum(field)` / `avg(field)` (the projection carries the field — there is no separate `*_star` call).
- Fetch via `DocumentSplitSums::fetch` / `DocumentSplitAverages::fetch`, fold with the overflow-safe `try_into_flat_map`, hex-encode keys, and serialize:
  - sum → `{"sums":{"<hex-key>":<i64>}}`
  - average → `{"averages":{"<hex-key>":{"count":<u64>,"sum":<i64>}}}` (caller divides)
- Empty `sum_property` is rejected up front with `InvalidParameter` (rs-sdk's select assertions require a non-empty field — fail fast for a clearer error).
- Three `where`/`order`/`limit` parsing helpers in `count.rs` promoted to `pub(super)` and reused, so cross-mode parsing stays identical across count/sum/average.
- **C ABI unchanged** (signatures already existed; only doc comments changed — cbindgen header auto-regenerates, byte-identical prototypes).
- Reclassified TEST_PLAN rows `DOC-13`/`DOC-14` `🚫 → 🔌` (FFI-available, no app UI yet).

## How Has This Been Tested?
- `cargo build -p rs-sdk-ffi` — clean.
- `cargo clippy -p rs-sdk-ffi --all-targets` — no new warnings.
- `cargo fmt --all` — no-op (already formatted).
- Confirmed the generated cbindgen prototypes are unchanged.

Follow-up (separate, build-gated): a SwiftUI Sum/Average view (mirroring `CountDocumentsView`) + a `summable`-index QA fixture to drive DOC-13/14 in-app (`🔌 → 🧪`).

## Breaking Changes
None — fills two previously-`NotImplemented` FFI bodies; ABI unchanged.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Average and sum aggregate query operations are now fully implemented and functional.

* **Tests**
  * Added unit tests to validate aggregate query results and expected output formats.

* **Documentation**
  * Updated test documentation to reflect the availability of average and sum query operations.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Review complete (commit d7237571)

File: packages/rs-sdk-ffi/src/document/queries/sum.rs [hunk 1]
```diff
@@ -56,22 +89,77 @@ pub unsafe extern "C" fn dash_sdk_document_sum(
...
+        // entries by summing over `in_key` with checked `i64`
+        // arithmetic; callers needing the unmerged shape should use a
+        // richer binding.
+        let flat_sums = DocumentSplitSums::fetch(&wrapper.sdk, sum_query)
+            .await
+            .map_err(|e| FFIError::InternalError(format!("Failed to fetch sum: {}", e)))?
+            .map(|s| s.try_into_flat_map())
+            .transpose()
+            .map_err(|e| FFIError::InternalError(format!("Failed to flatten sum result: {}", e)))?
+            .unwrap_or_default();
```
Comment (thepastaclaw) [inline packages/rs-sdk-ffi/src/document/queries/sum.rs]: **🟡 Suggestion: Preserve typed SDK errors from the sum/average fetch instead of collapsing into InternalError**

DocumentSplitSums::fetch and DocumentSplitAverages::fetch return dash_sdk::Error. packages/rs-sdk-ffi/src/error.rs:102-188 already classifies that variant into NetworkError (DapiClientError/NoAvailableAddressesToRetry), Timeout (TimeoutReached), DriveInternalError, NotFound, ProtocolError, etc. The current `map_err(|e| FFIError::InternalError(format!("Failed to fetch sum: {}", e)))` (sum.rs:140 and average.rs:157) discards that classification, so every transient transport failure or proof error surfaces to Swift as code 99 (InternalError) instead of the typed code that Swift can act on. Using `.map_err(FFIError::from)?` lets the existing `From<dash_sdk::Error> for FFIError` (#[from] on SDKError) feed the existing classifier. Note the existing count entry point has the same shortcut — aligning the three call sites together (or in a small follow-up) avoids future divergence.

<sub>source: ['codex']</sub>

File: packages/rs-sdk-ffi/src/document/queries/sum.rs [hunk 2]
```diff
@@ -56,22 +89,77 @@ pub unsafe extern "C" fn dash_sdk_document_sum(
...
+
+        let sum_property_str = CStr::from_ptr(sum_property)
+            .to_str()
+            .map_err(FFIError::from)?;
+        if sum_property_str.is_empty() {
+            return Err(FFIError::InvalidParameter(
+                "sum_property must name the integer property to sum; got an empty string"
+                    .to_string(),
+            ));
+        }
```
Comment (thepastaclaw) [inline packages/rs-sdk-ffi/src/document/queries/sum.rs]: **🟡 Suggestion: Add small unit tests for the new sum/average FFI surface (empty sum_property, JSON shape, hex keys)**

The PR turns two stable NotImplemented stubs into a real C ABI surface but adds no rs-sdk-ffi tests for the new code. The empty-`sum_property` rejection (sum.rs:112-117 and average.rs:129-134) is pure FFI logic — no SDK / runtime / contract needed — and is the only new validation step; a unit test that constructs an empty NUL-terminated string and asserts FFIError::InvalidParameter (mirroring the style of the existing decode_ffi_limit tests at the bottom of count.rs) would lock the contract in cheaply. The DocumentSumResult / DocumentAverageResult JSON shapes (`{"sums":{...}}`, `{"averages":{"<hex>":{"count":..,"sum":..}}}`) and the hex-key encoding are also pure data-shaping and equally cheap to round-trip. End-to-end proof-verified coverage via the planned `summable` index fixture is reasonably deferred; the FFI-shape tests are not.

<sub>source: ['claude', 'codex']</sub>

File: packages/rs-sdk-ffi/src/document/queries/sum.rs [hunk 3]
```diff
@@ -39,11 +66,17 @@ use std::os::raw::c_char;
...
+///   `> 0` for an explicit cap, `0` / `< -1` rejected. See
+///   [`super::count::dash_sdk_document_count`] for the full contract.
 ///
 /// # Safety
 /// Same contract as [`super::count::dash_sdk_document_count`]. All
 /// pointers must be valid for the duration of the call.
+/// - `sdk_handle` and `data_contract_handle` must be valid, non-null pointers.
+/// - `document_type` and `sum_property` must be NUL-terminated C strings valid for the duration of the call.
+/// - `where_json`, `order_by_json`, and `group_by_json` may be null; if non-null they must be NUL-terminated JSON strings.
+/// - On success, returns a heap-allocated C string pointer; caller must free it using SDK routines.
```
Comment (thepastaclaw) [inline packages/rs-sdk-ffi/src/document/queries/sum.rs:130]: **💬 Nitpick: `# Safety` section states the contract twice (intra-doc reference + inline bullets)**

Both sum.rs:73-79 and average.rs:90-96 keep the one-line `Same contract as [...]` prose and immediately follow it with a four-bullet per-parameter list saying the same thing in more detail. The two halves can drift independently if count's / sum's contract is tightened later. Prefer the inline bullets only — cbindgen-generated headers don't follow Rust intra-doc links, so the bullets are what FFI consumers actually see.

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Clean FFI implementation of dash_sdk_document_sum / dash_sdk_document_average mirroring the existing dash_sdk_document_count pattern: shared parameter-parsing helpers are promoted to pub(super) and reused verbatim, null-pointer guards and UTF-8 validation are consistent, and the (count, sum) result shape is correctly handled via try_into_flat_map. No blocking correctness, safety, FFI ABI, or consensus issues found. The only worthwhile suggestions are a typed-SDK-error preservation tweak and a small unit-test gap for the new entry points.

🟡 2 suggestion(s) | 💬 1 nitpick(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `packages/rs-sdk-ffi/src/document/queries/sum.rs`:
- [SUGGESTION] packages/rs-sdk-ffi/src/document/queries/sum.rs:138-144: Preserve typed SDK errors from the sum/average fetch instead of collapsing into InternalError
  DocumentSplitSums::fetch and DocumentSplitAverages::fetch return dash_sdk::Error. packages/rs-sdk-ffi/src/error.rs:102-188 already classifies that variant into NetworkError (DapiClientError/NoAvailableAddressesToRetry), Timeout (TimeoutReached), DriveInternalError, NotFound, ProtocolError, etc. The current `map_err(|e| FFIError::InternalError(format!("Failed to fetch sum: {}", e)))` (sum.rs:140 and average.rs:157) discards that classification, so every transient transport failure or proof error surfaces to Swift as code 99 (InternalError) instead of the typed code that Swift can act on. Using `.map_err(FFIError::from)?` lets the existing `From<dash_sdk::Error> for FFIError` (#[from] on SDKError) feed the existing classifier. Note the existing count entry point has the same shortcut — aligning the three call sites together (or in a small follow-up) avoids future divergence.
- [SUGGESTION] packages/rs-sdk-ffi/src/document/queries/sum.rs:112-117: Add small unit tests for the new sum/average FFI surface (empty sum_property, JSON shape, hex keys)
  The PR turns two stable NotImplemented stubs into a real C ABI surface but adds no rs-sdk-ffi tests for the new code. The empty-`sum_property` rejection (sum.rs:112-117 and average.rs:129-134) is pure FFI logic — no SDK / runtime / contract needed — and is the only new validation step; a unit test that constructs an empty NUL-terminated string and asserts FFIError::InvalidParameter (mirroring the style of the existing decode_ffi_limit tests at the bottom of count.rs) would lock the contract in cheaply. The DocumentSumResult / DocumentAverageResult JSON shapes (`{"sums":{...}}`, `{"averages":{"<hex>":{"count":..,"sum":..}}}`) and the hex-key encoding are also pure data-shaping and equally cheap to round-trip. End-to-end proof-verified coverage via the planned `summable` index fixture is reasonably deferred; the FFI-shape tests are not.
```
</details>

File: packages/rs-sdk-ffi/src/document/queries/sum.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/sum.rs]: Done in 762f68d — the `.fetch(...)` errors now use `.map_err(FFIError::from)?` in sum/average **and** count, so they route through the existing `SDKError` classifier (NetworkError/Timeout/DriveInternalError/…) instead of collapsing to InternalError. The `try_into_flat_map` overflow stays InternalError (data error, not transport).

File: packages/rs-sdk-ffi/src/document/queries/sum.rs (hunk 2 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/sum.rs]: Done in 762f68d — extracted `validate_aggregation_property` (pure, used by both sum and average) and added unit tests: empty-field → InvalidParameter, plus DocumentSumResult/DocumentAverageResult JSON-shape round-trips (mirroring the decode_ffi_limit test style). `cargo test -p rs-sdk-ffi` green.

File: packages/rs-sdk-ffi/src/document/queries/sum.rs (hunk 3 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/sum.rs:130]: Done in 762f68d — dropped the redundant "Same contract as […]" Safety prose in sum.rs/average.rs, keeping the per-parameter bullet list.

File: packages/rs-sdk-ffi/src/document/queries/sum.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/sum.rs]: (retry) Done in 762f68d — fetch errors now use .map_err(FFIError::from) in sum/average/count, routing through the existing SDKError classifier instead of InternalError.

File: packages/swift-sdk/SwiftExampleApp/TEST_PLAN.md [hunk 4]
```diff
@@ -215,8 +215,8 @@ The app is a full multi-wallet client: `PlatformWalletManager` holds N wallets c
...
-| DOC-13 | Aggregation — sum of a numeric property | Platform | Uncommon | 🚫 | FFI `dash_sdk_document_sum` returns `NotImplemented` — blocked upstream on grovedb PR 670 (range/sum aggregate). Will need a `summable` index once unblocked. |
-| DOC-14 | Aggregation — average of a numeric property | Platform | Uncommon | 🚫 | FFI `dash_sdk_document_average` returns `NotImplemented` — blocked upstream on grovedb PR 670. Will need a `summable` index once unblocked. |
+| DOC-13 | Aggregation — sum of a numeric property | Platform | Uncommon | 🔌 | FFI `dash_sdk_document_sum` **implemented** (the grovedb PR 670 aggregate-sum capability is present; wraps rs-sdk `DocumentSplitSums::fetch`, proof-verified) → `{sums:{hexKey:i64}}`. No app UI yet; needs a contract doc type with a `summable` index on a numeric property. |
+| DOC-14 | Aggregation — average of a numeric property | Platform | Uncommon | 🔌 | FFI `dash_sdk_document_average` **implemented** (wraps rs-sdk `DocumentSplitAverages::fetch`) → `{averages:{hexKey:{count,sum}}}` (caller divides). No app UI yet; needs a `countable`+`summable` index. |
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/swift-sdk/SwiftExampleApp/TEST_PLAN.md:219]: Fixed in 71440d7d. Appendix A getDocuments row → `✅ / 🧪 / 🔌` and "SUM/AVG are FFI-available (`DOC-13/14`) — no app UI yet". In Appendix B, moved DOC-13/14 out of "🚫 Not implemented anywhere" into "🔌 SDK-only (FFI/wrapper exists, no UI)"; GRP-04 stays under 🚫. Good catch on the stale cross-refs.

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

All three prior reviewer findings (typed-error preservation, FFI unit tests, duplicated # Safety prose) are resolved in the current head. One valid CodeRabbit doc inconsistency remains: TEST_PLAN.md's Appendix A and Appendix B still describe DOC-13/DOC-14 as upstream-blocked stubs, contradicting the §4.7 rows and the now-implemented FFI surface this PR ships.

Review coverage note: the following reviewer lanes failed or produced unparseable output and were not included in validated findings: claude-ffi-engineer, claude-general, claude-rust-quality, claude-security-auditor. In this run the Claude ACP lanes failed with the current account limit resetting at 8am America/Chicago; available Codex lanes were still verified.

Reviewed commit `762f68d0` with cumulative reconciliation against prior commit `074c91bc`.

### Findings

- **[suggestion] Appendix A still says SUM/AVG are upstream-blocked; Appendix B still lists them as NotImplemented stubs** — `packages/swift-sdk/SwiftExampleApp/TEST_PLAN.md:394`
  The §4.7 catalog rows for DOC-13/DOC-14 (lines 218–219) correctly mark sum/average as 🔌 with `dash_sdk_document_sum` / `dash_sdk_document_average` implemented, but two cross-references elsewhere in the same document still describe them as upstream-blocked stubs:

- Line 394 (Appendix A — gRPC read-RPC coverage, Document section): `SUM/AVG are upstream-blocked (grovedb PR 670) — DOC-13/14.`
- Lines 481–482 (Appendix B — Theoretically possible but not runnable in-app): `DOC-13 document SUM aggregation — FFI stub returns NotImplemented (blocked on grovedb PR 670)` / `DOC-14 document AVERAGE aggregation — FFI stub returns NotImplemented (blocked on grovedb PR 670)`.

Those references contradict the rows this PR just updated and falsely advertise the stub behavior the PR removed. Update Appendix A to reflect that SUM/AVG are now implemented (FFI-only, no in-app UI) and remove DOC-13/DOC-14 from Appendix B (or move them to the SDK-implemented / no-UI bucket).

### Resolved prior findings
- **Preserve typed SDK errors from the sum/average fetch instead of collapsing into InternalError** — Fixed at sum.rs:151–153 and average.rs:149–151 — both now use `.map_err(FFIError::from)?` on the SDK `fetch` call, preserving the existing `From<dash_sdk::Error>` classification (NetworkError / Timeout / DriveInternalError / NotFound / ProtocolError). The adjacent count.rs path was aligned in the same commit.
- **Add small unit tests for the new sum/average FFI surface (empty sum_property, JSON shape, hex keys)** — Fixed by extracting the empty-property rejection into `validate_aggregation_property` (sum.rs:38–47) and adding pure-FFI tests: `validate_aggregation_property_rejects_empty_accepts_named`, `document_sum_result_serializes_to_expected_shape` (covers empty + hex keys, signed value), and `document_average_result_serializes_to_expected_shape` (count+sum pair shape) at sum.rs:180–220 and average.rs:180–201.
- **`# Safety` section states the contract twice (intra-doc reference + inline bullets)** — Fixed — sum.rs:93–97 and average.rs:91–95 now keep only the four-bullet per-parameter contract; the `Same contract as …` intra-doc reference prose was removed.

_Inline posting hit GitHub HTTP 422, so I posted the same verified findings as a top-level review body._

**Comment by QuantumExplorer** [comment]
Heads-up for reviewers/mergers: the one open finding (Appendix A/B still describing DOC-13/14 as grovedb-blocked stubs) was reviewed against `762f68d0` but is already fixed in the current head `71440d7d` — Appendix A now reads `✅ / 🧪 / 🔌` + "FFI-available … no app UI yet", and DOC-13/14 were moved into Appendix B's "🔌 SDK-only" bucket (CodeRabbit confirmed this fix and its thread is resolved). The remaining red checks are the pre-existing `sdk_builder_default_seeds_atomic_to_floor` macOS flake in rs-sdk (a crate this PR doesn't touch; fix in flight) and the codecov drop that's a downstream consequence of that aborted Rust coverage upload — not a regression from this change.

File: packages/rs-sdk-ffi/src/document/queries/sum.rs [hunk 5]
```diff
@@ -56,22 +107,114 @@ pub unsafe extern "C" fn dash_sdk_document_sum(
...
+        // entries by summing over `in_key` with checked `i64`
+        // arithmetic; callers needing the unmerged shape should use a
+        // richer binding.
+        let flat_sums = DocumentSplitSums::fetch(&wrapper.sdk, sum_query)
+            .await
+            .map_err(FFIError::from)?
+            .map(|s| s.try_into_flat_map())
+            .transpose()
+            .map_err(|e| FFIError::InternalError(format!("Failed to flatten sum result: {}", e)))?
+            .unwrap_or_default();
```
Comment (thepastaclaw) [inline packages/rs-sdk-ffi/src/document/queries/sum.rs]: **🔴 Blocking: Ungrouped SUM queries don't always return the single-empty-key aggregate the doc promises**

The FFI doc-comments (lines 53-55 and 66-67) state that for empty/null `group_by_json` the result is a one-entry map with an empty key (`sums[""]` is the total). However the path always calls `DocumentSplitSums::fetch`, which `rs-sdk/src/platform/documents/document_split_sums.rs:5-8` explicitly scopes to `select=SUM, group_by=[...]` and points aggregate callers at `DocumentSum`. Per `sum_proof_helpers.rs:204-292`, only `DocumentSumMode::RangeProof` (range, no In, no distinct) emits the single-empty-key entry the doc promises; `PointLookupProof` (point/`in`) and `RangeAggregateCarrierProof` (`in + range`) emit one entry per matched key, and `try_into_flat_map` preserves that shape. So an iOS caller passing `where field in [...]` with no `group_by_json` gets `{sums:{<hexKey1>:v1,<hexKey2>:v2,...}}` (or `{}` for no matches) instead of the documented `{sums:{"":<total>}}`. Either route ungrouped queries through `dash_sdk::platform::documents::document_sum::DocumentSum::fetch` (which folds all verified entries into a single `i64` with checked-add overflow handling and returns 0 for empty aggregates), or correct the doc-comment to describe the actual per-key wire shape.

<sub>source: ['codex']</sub>

File: packages/rs-sdk-ffi/src/document/queries/average.rs [hunk 6]
```diff
@@ -61,23 +105,97 @@ pub unsafe extern "C" fn dash_sdk_document_average(
...
+        // unmerged shape should use a richer binding.
+        let flat_averages = DocumentSplitAverages::fetch(&wrapper.sdk, average_query)
+            .await
+            .map_err(FFIError::from)?
+            .map(|s| s.try_into_flat_map())
+            .transpose()
+            .map_err(|e| {
+                FFIError::InternalError(format!("Failed to flatten average result: {}", e))
+            })?
+            .unwrap_or_default();
```
Comment (thepastaclaw) [inline packages/rs-sdk-ffi/src/document/queries/average.rs]: **🔴 Blocking: Ungrouped AVG queries don't always return the single-empty-key aggregate the doc promises**

Symmetric to the sum issue: the FFI documents ungrouped average results as a single `averages[""]` `(count, sum)` pair, but the implementation always calls `DocumentSplitAverages::fetch` and flattens via `try_into_flat_map`. `rs-sdk`'s `DocumentSplitAverages` is the group-by surface; aggregate callers should use `DocumentAverage::fetch` (which folds all verified entries into one `(count, sum)` pair and returns `(0, 0)` for empty aggregates). Under `PointLookupProof` and `RangeAggregateCarrierProof` shapes, an ungrouped average request returns one `(count, sum)` pair per matched key (or `{}`), so callers computing `sum/count` from the documented `averages[""]` get the wrong number or a missing key. Either route ungrouped queries through `DocumentAverage::fetch` or correct the doc-comment to describe the actual per-key wire shape.

<sub>source: ['codex']</sub>

**Review by thepastaclaw — CHANGES_REQUESTED** [review]
## Code Review

Prior Appendix A/B inconsistency finding is FIXED in the current head (line 394 + Appendix B reclassification). New cumulative review found two convergent correctness issues in the sum/average FFI: ungrouped requests are routed through `DocumentSplitSums`/`Averages`, which the rs-sdk source documents as the group-by path — for `In`-based and carrier-aggregate where clauses this returns per-key entries instead of the documented single empty-key total. Other agent findings (error-code classification in shared parse helpers, panic-safety) are pre-existing or out of scope for this PR.

🔴 2 blocking

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `packages/rs-sdk-ffi/src/document/queries/sum.rs`:
- [BLOCKING] packages/rs-sdk-ffi/src/document/queries/sum.rs:144-157: Ungrouped SUM queries don't always return the single-empty-key aggregate the doc promises
  The FFI doc-comments (lines 53-55 and 66-67) state that for empty/null `group_by_json` the result is a one-entry map with an empty key (`sums[""]` is the total). However the path always calls `DocumentSplitSums::fetch`, which `rs-sdk/src/platform/documents/document_split_sums.rs:5-8` explicitly scopes to `select=SUM, group_by=[...]` and points aggregate callers at `DocumentSum`. Per `sum_proof_helpers.rs:204-292`, only `DocumentSumMode::RangeProof` (range, no In, no distinct) emits the single-empty-key entry the doc promises; `PointLookupProof` (point/`in`) and `RangeAggregateCarrierProof` (`in + range`) emit one entry per matched key, and `try_into_flat_map` preserves that shape. So an iOS caller passing `where field in [...]` with no `group_by_json` gets `{sums:{<hexKey1>:v1,<hexKey2>:v2,...}}` (or `{}` for no matches) instead of the documented `{sums:{"":<total>}}`. Either route ungrouped queries through `dash_sdk::platform::documents::document_sum::DocumentSum::fetch` (which folds all verified entries into a single `i64` with checked-add overflow handling and returns 0 for empty aggregates), or correct the doc-comment to describe the actual per-key wire shape.

In `packages/rs-sdk-ffi/src/document/queries/average.rs`:
- [BLOCKING] packages/rs-sdk-ffi/src/document/queries/average.rs:149-157: Ungrouped AVG queries don't always return the single-empty-key aggregate the doc promises
  Symmetric to the sum issue: the FFI documents ungrouped average results as a single `averages[""]` `(count, sum)` pair, but the implementation always calls `DocumentSplitAverages::fetch` and flattens via `try_into_flat_map`. `rs-sdk`'s `DocumentSplitAverages` is the group-by surface; aggregate callers should use `DocumentAverage::fetch` (which folds all verified entries into one `(count, sum)` pair and returns `(0, 0)` for empty aggregates). Under `PointLookupProof` and `RangeAggregateCarrierProof` shapes, an ungrouped average request returns one `(count, sum)` pair per matched key (or `{}`), so callers computing `sum/count` from the documented `averages[""]` get the wrong number or a missing key. Either route ungrouped queries through `DocumentAverage::fetch` or correct the doc-comment to describe the actual per-key wire shape.
```
</details>

File: packages/rs-sdk-ffi/src/document/queries/sum.rs (hunk 5 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/sum.rs]: Fixed in d723757 — ungrouped SUM now routes through `DocumentSum::fetch` (folds all verified entries to one i64) → `{sums:{"":<total>}}` (0 when the aggregate is absent); grouped keeps `DocumentSplitSums` + `try_into_flat_map` (per-group hex keys). Doc-comments now describe both shapes and 4 ungrouped-shape tests were added.

File: packages/rs-sdk-ffi/src/document/queries/average.rs (hunk 6 above)
Comment (QuantumExplorer) [reply packages/rs-sdk-ffi/src/document/queries/average.rs]: Fixed in d723757 — symmetric to sum: ungrouped AVG now routes through `DocumentAverage::fetch` (one folded `{count,sum}`) → `{averages:{"":{count,sum}}}` ((0,0) when absent); grouped keeps `DocumentSplitAverages` per-group.

**Review by thepastaclaw — APPROVED** [review]
## Code Review

Both prior blocking findings (ungrouped SUM/AVG routed through the per-group DocumentSplitSums/Averages type instead of the aggregate type) are FIXED in the current head. Verified at sum.rs:179-194 and average.rs:170-186 — the ungrouped path now branches on group_by.is_empty() and routes through DocumentSum::fetch / DocumentAverage::fetch, returning the documented single empty-key total with zero fallback. Grouped queries retain DocumentSplitSums/Averages with try_into_flat_map. No new in-scope blocking, suggestion, or nitpick findings from any reviewer lane; CodeRabbit reported zero inline findings (coverage was rate-limited and degraded).
